orchagent/saihelper.cpp

```diff
@@ -286,9 +291,18 @@ sai_status_t initSaiPhyApi(swss::gearbox_phy_t *phy)
     attr.value.u32 = 0;
     attrs.push_back(attr);
 
+    if (phy->hwinfo.length() > HWINFO_MAX_SIZE)
```

Comment:
if i understand correctly, the comparison is really for interface_name part, it is not the combined string. for example, you can have interface_name > ifnamsiz, but phy_id less than 5 bytes, so in total it is still less than HWINFO_MAC_SIZE, but the phy driver will see error since the interface_name exceed the limit?

Reply:
That's a good point. I only thought about memory constraints, not about sanity of the data. That's why the check is done this way. I'll post a new version with a proper sanity check for the sub components of hwinfo.